File: client-clojure/src/sirqul_io_t_platform/specs/contact_info_response.clj
(ns sirqul-io-t-platform.specs.contact-info-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.address-response :refer :all]
            [sirqul-io-t-platform.specs.cell-carrier-response :refer :all]
            )
  (:import (java.io File)))


(def contact-info-response-data
  {
   (ds/opt :homePhone) string?
   (ds/opt :cellPhone) string?
   (ds/opt :cellPhoneCarrier) string?
   (ds/opt :businessPhone) string?
   (ds/opt :businessPhoneExt) string?
   (ds/opt :faxNumber) string?
   (ds/opt :timeZone) string?
   (ds/opt :utcOffset) string?
   (ds/opt :code501c3) string?
   (ds/opt :emailAddress) string?
   (ds/opt :address) address-response-spec
   (ds/opt :web) string?
   (ds/opt :cellCarrier) cell-carrier-response-spec
   })

(def contact-info-response-spec
  (ds/spec
    {:name ::contact-info-response
     :spec contact-info-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/ContactInfoResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ContactInfoResponse._

case class ContactInfoResponse (
  homePhone: Option[String],
cellPhone: Option[String],
cellPhoneCarrier: Option[String],
businessPhone: Option[String],
businessPhoneExt: Option[String],
faxNumber: Option[String],
timeZone: Option[String],
utcOffset: Option[String],
code501c3: Option[String],
emailAddress: Option[String],
address: Option[AddressResponse],
web: Option[String],
cellCarrier: Option[CellCarrierResponse])

object ContactInfoResponse {
  import DateTimeCodecs._

  implicit val ContactInfoResponseCodecJson: CodecJson[ContactInfoResponse] = CodecJson.derive[ContactInfoResponse]
  implicit val ContactInfoResponseDecoder: EntityDecoder[ContactInfoResponse] = jsonOf[ContactInfoResponse]
  implicit val ContactInfoResponseEncoder: EntityEncoder[ContactInfoResponse] = jsonEncoderOf[ContactInfoResponse]
}
